internal/mesh/internal/controllers/implicitdestinations/controller.go

```diff
@@ -0,0 +1,314 @@
+// - only ports exposed by CR or CTP explicitly
+
+/*
+Data Relationships:
+
+Reconcile:
+- read WI[source] (ignore)
+- list CTPs by WI[source]
+  - turn CTP.id -> WI[backend].id
+  - list SVCs by WI[backend]
+    - list CRs by SVC[backend]
+      - turn CR.id -> SVC[dest].id
+	  - emit SVC[dest]
+
+DepMappers:
+- CR <list> SVC[backend] <list> WI[backend] <align> CTP <list> WI[source] <align> CID
+-           SVC[backend] <list> WI[backend] <align> CTP <list> WI[source] <align> CID
+-                                                   CTP <list> WI[source] <align> CID
+- bound refs for all
+
+*/
+
+func Controller(globalDefaultAllow bool) *controller.Controller {
+	m := &mapAndTransformer{globalDefaultAllow: globalDefaultAllow}
+
+	boundRefsMapper := dependency.CacheListMapper(pbmesh.ComputedImplicitDestinationsType, boundRefsIndex.Name())
+
+	return controller.NewController(ControllerID,
+		pbmesh.ComputedImplicitDestinationsType,
+		boundRefsIndex,
+	).
+		WithWatch(pbauth.WorkloadIdentityType,
+			// BoundRefs: none
+			dependency.ReplaceType(pbmesh.ComputedImplicitDestinationsType),
+		).
+		WithWatch(pbauth.ComputedTrafficPermissionsType,
+			// BoundRefs: the WI source refs are interior up-pointers and may change.
+			dependency.MultiMapper(boundRefsMapper, m.MapComputedTrafficPermissions),
+			ctpBySourceWorkloadIdentityIndex,
+			ctpByWildcardSourceIndexCreator(globalDefaultAllow),
+		).
+		WithWatch(pbcatalog.ServiceType,
+			// BoundRefs: the WI slice in the status conds is an interior up-pointer and may change.
+			dependency.MultiMapper(boundRefsMapper, m.MapService),
+			serviceByWorkloadIdentityIndex,
+		).
+		WithWatch(pbmesh.ComputedRoutesType,
+			// BoundRefs: the backend services are interior up-pointers and may change.
+			dependency.MultiMapper(boundRefsMapper, m.MapComputedRoutes),
+			computedRoutesByBackendServiceIndex,
+		).
+		WithReconciler(&reconciler{
+			defaultAllow: globalDefaultAllow,
+		})
+}
+
+type reconciler struct {
+	defaultAllow bool
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, rt controller.Runtime, req controller.Request) error {
+	rt.Logger = rt.Logger.With("resource-id", req.ID, "controller", ControllerID)
+
+	wi := resource.ReplaceType(pbauth.WorkloadIdentityType, req.ID)
+
+	workloadIdentity, err := cache.GetDecoded[*pbauth.WorkloadIdentity](rt.Cache, pbauth.WorkloadIdentityType, "id", wi)
+	if err != nil {
+		rt.Logger.Error("error retrieving corresponding Workload Identity", "error", err)
+		return err
+	} else if workloadIdentity == nil {
+		rt.Logger.Trace("workload identity has been deleted")
+		return nil
+	}
+
+	// generate new CID and compare, if different, write new one, if not return without doing anything
+	newData, err := r.generateComputedImplicitDestinations(ctx, rt, wi)
+	if err != nil {
+		rt.Logger.Error("error generating computed implicit destinations", "error", err)
+		// TODO: update the workload identity with this error as a status condition?
```

Comment:
I carried this comment forward from the prior branch. Did we want to do this? I don't recall if we follow this pattern for computed resources across the board at the moment.